### DISCUSSION TITLE
Color background for checkmark widgets that are ticked already

### DISCUSSION THREAD

**Discussion by AriApar** [iSoron/uhabits#615] [Feature Requests]
First of all, amazing work with this app, been using it for some time now and am amazed at how integral it has become for my phone usage.

I had a small UX suggestion on the checkmark widgets. A habit can either be explicitly checked by a user, implicitly checked if the user already fulfilled the number of times they needed to check the habit in the last X days, or unchecked. When a feature is explicitly checked, the widget background also changes color. I was wondering if it'd make sense for an implicitly checked widget to also have some color (perhaps with some transparency)

![Screenshot_20200803-212530__01](https://user-images.githubusercontent.com/4266136/89215518-779ed300-d5d1-11ea-9d3b-d05cd5ef15d3.jpg)

In the screenshot above, the ~blog post~ (edit: this should have said "friends&family", thanks for the correction @wobbba) widget would be the one I'm talking about, and it'd be nice to look at my screen and immediately distinguish between the colorful vs grey boxes, to see which habits I should do today and which are already done (checkmark already provides this but is less in-your-face)

Thanks for considering, and again, amazing work with your app!

**Comment by KristianTashkov** [comment]
To be honest I have the same consideration for the checkmarks in the app as well. On a dimmed screen you can't make the distinction(implicit check or unchecked) at a glance. It makes sense "color" to always mean success and playing with opacity to show the difference between explicit and implicit.

**Comment by end-ichi** [comment]
> I was wondering if it'd make sense for an implicitly checked widget to also have some color (perhaps with some transparency)

Came here to suggest this, so +1, that would be great!!

> In the screenshot above, the blog post widget would be the one I'm talking about

I'm confused, the blog post one isn't implicitly checked, it's the "friends & family" one, that one should have some colour IMO.

I also think it would be cool if the transparency and/or saturation and/or brightness of the colour would change according to how far away you are from breaking the chain. 

E.g. if it's a once a week habit and you just did it yesterday the colour is at its brightest/most saturated/most opaque and slowly fades day by day until it's at its most dark/unsaturated/transparent on the last implicitly checked day and then one day later it goes to completely colourless with an X, the way it is now.

(EDIT: much rather than the following, I'd like the basic functionality be actually implemented in any way)
I would also love to see some heavy customization potential in the settings, e.g. min and max transparency/saturation/brightness, which of the three or multiple, maybe even functions of the way it fades (linear, exponential, etc.), possibly even let the user input a custom function. Or customize how to handle short time windows: if it's a once per two days habit, should the colour be vibrant because it was completed yesterday or bleak because it's the last implicit day (I'd say the latter but why not let the user decide?) Similarly for 3 days and so on.

Anyway, the most important is that there's partial colour at all because as you said seeing it at a glance would be better and there is just something so satisfying about a wall of coloured habit-tiles :)

**Comment by tristanlabelle** [comment]
I would also like this. I just created a bunch of weekly tasks (music practice) and an entire home screen of widgets so that I would progressively check them all during the week. The current style of "completed on a previous day" task is difficult to distinguish from the uncompleted state. For my use case, I don't want to distinguish between completion on the same or a previous day and would like to have the widget keep its bright color until the end of the week. Otherwise I'll end up re-checking them every day just so I can better see what's done and what's not.

Really great app btw!

**Comment by cjseaslug** [comment]
+1 — been using Loop for about a week and its already helped me develop a really good routine for critical daily tasks! It's seeming to be one thing that really works for my ADHD. Thank you for making this!

I genuinely would love ticked tasks to remaim highlighted despite a day reset, this would make the "loop" a lot tighter and intuitively actionable for me ;) although personally, i wouldnt want any opacity other than 100, like other tasks completed in the same day.

Sorry to ping directly, but I wanted to check since this was asked in 2020; is there any possibility of this becoming an option @iSoron?

**Comment by iSoron** [comment]
I agree that the current colors are not consistent. Below is a possible redesign. PRs implementing it are very welcome.

### Current
![Screenshot_1647933866 1](https://user-images.githubusercontent.com/4699657/159429874-ca865cdf-bf13-4ac4-b44e-89ab44fe0fe4.png)

### Proposed
![Group 1](https://user-images.githubusercontent.com/4699657/159429892-38aac3ff-9560-45bb-aa6c-e9738e730a69.png)

**Reply by end-ichi** [reply]
Thank you!! I like it. This is basically what I wanted. 
Mabe it would be even better if Habit 3 was slightly transparent to better capture its "in-between-ness" between Habit 1 (coloured, opaque) and Habit 2 (uncoloured, transparent) but that's not super-important.
I'm sorry, I still haven't taught myself mobile development yet otherwise I would've tried to code it myself a while ago...

**Reply by cjseaslug** [reply]
Thank you so much for your blessing on the PR and the mockup, I definitely can't make any promises but know a little Java and may take a crack at it.

@wobbba just ran into a good use case for your "opacity decay" idea with exercise/gym stuff - ideally I'd want to see this as a per task option for things that can be completed on a looser schedule and can be repeated before they expire, but I'm using my widgets for a once weekly injection and a once-every-3-days medication, so keeping full opacity makes the most sense for that use :0 What do you think?

**Comment by rosie228** [comment]
Any update on this becoming a thing? Would love to see it happen, I have a whole screen of checkmark widgets and would love a wall of colour until it resets at the end of each week!

**Reply by end-ichi** [reply]
I've created a pull request that implements this, with any luck we will get it in the upcoming v2.2.0 🤞
(the most basic version anyway, I am new to mobile App development)

**Reply by patrik-bartak** [reply]
I was just about to implement this for myself but great that someone already did it. I do not see it in 2.2.1 though. Any idea when this will be available or what needs to be done?

**Reply by end-ichi** [reply]
I don't know. Sorry for direct ping, but could we maybe get a minor release for this or any info on when we can expect the next release @iSoron ? It's been a while 😅
